ops/internal/config/chain.go

```diff
@@ -6,45 +6,47 @@ import (
-	SuperchainTime       *uint64      `toml:"superchain_time"`
-	DataAvailabilityType string       `toml:"data_availability_type"`
-	DeploymentTxHash     *common.Hash `toml:"deployment_tx_hash"`
+type StagedChain struct {
+	Chain
+	ShortName                    string             `toml:"-"`
+	Superchain                   Superchain         `toml:"superchain"`
+	BaseFeeVaultRecipient        ChecksummedAddress `toml:"base_fee_vault_recipient"`
+	L1FeeVaultRecipient          ChecksummedAddress `toml:"l1_fee_vault_recipient"`
+	SequencerFeeVaultRecipient   ChecksummedAddress `toml:"sequencer_fee_vault_recipient"`
+	DeploymentTxHash             *common.Hash       `toml:"deployment_tx_hash"`
+	DeploymentL1ContractsVersion *artifacts.Locator `toml:"deployment_l1_contracts_version"`
+	DeploymentL2ContractsVersion *artifacts.Locator `toml:"deployment_l2_contracts_version"`
 }
 
 type Chain struct {
-	Metadata
-	ChainID           uint64              `toml:"chain_id"`
-	BatchInboxAddr    *ChecksummedAddress `toml:"batch_inbox_addr"`
-	BlockTime         uint64              `toml:"block_time"`
-	SeqWindowSize     uint64              `toml:"seq_window_size"`
-	MaxSequencerDrift uint64              `toml:"max_sequencer_drift"`
-	GasPayingToken    *ChecksummedAddress `toml:"gas_paying_token"`
-	Hardforks         Hardforks           `toml:"hardforks"`
-	Optimism          Optimism            `toml:"optimism"`
-	AltDA             *AltDA              `toml:"alt_da"`
-	Genesis           Genesis             `toml:"genesis"`
-	Roles             Roles               `toml:"roles"`
-	Addresses         Addresses           `toml:"addresses"`
+	Name                 string              `toml:"name"`
+	PublicRPC            string              `toml:"public_rpc"`
+	SequencerRPC         string              `toml:"sequencer_rpc"`
+	Explorer             string              `toml:"explorer"`
+	SuperchainLevel      int                 `toml:"superchain_level"`
+	GovernedByOptimism   bool                `toml:"governed_by_optimism"`
+	SuperchainTime       *uint64             `toml:"superchain_time"`
+	DataAvailabilityType string              `toml:"data_availability_type"`
+	ChainID              uint64              `toml:"chain_id"`
+	BatchInboxAddr       *ChecksummedAddress `toml:"batch_inbox_addr"`
+	BlockTime            uint64              `toml:"block_time"`
+	SeqWindowSize        uint64              `toml:"seq_window_size"`
+	MaxSequencerDrift    uint64              `toml:"max_sequencer_drift"`
+	GasPayingToken       *ChecksummedAddress `toml:"gas_paying_token,omitempty"`
+	Hardforks            Hardforks           `toml:"hardforks"`
+	Optimism             Optimism            `toml:"optimism"`
+	AltDA                *AltDA              `toml:"alt_da"`
+	Genesis              Genesis             `toml:"genesis"`
+	Roles                Roles               `toml:"roles"`
+	Addresses            Addresses           `toml:"addresses"`
```

Comment:
Would it be better to rename from `addresses` --> `contracts`? Also curious if `BatchInboxAddr` could be moved inside of this struct, or within `roles` instead of staying as a top level field

Reply:
Changing these would be a much bigger refactor. The monorepo contracts and superchain-ops both assume an addresses stanza.

BatchInboxAddr is a special case - it's more of a protocol constant than it is a live address with a contract deployed on it, so it doesn't really fit Addresses or Roles. Unless you're strongly opinionated about it I'd rather keep it where it is.